### ISSUE TITLE
Add field descriptor to set column name in schema to something else

### ISSUE THREAD

**Issue by Coteh** [eveningkid/denodb#41]
I think it would be nice to have a field descriptor called `fieldName` or something similiar that lets you set the name of a field to something else in the table schema. This would be especially useful in cases where you have something like `userName` for example which is camelCase and is typically convention for field names in JS/TS but not usually convention for field names in SQL (from my understanding, snake_case is SQL convention). I understand that one of the pillars of an ORM is to abstract away SQL queries, but it would be nice to have some sort of way to keep things consistent across both your code and SQL schemas.

**Comment by eveningkid** [comment]
Hello James,

I agree and was actually thinking about this yesterday!

I don't think we should add a field descriptor for now but instead, encourage users to keep it javascript-like on the ORM side but SQL-like on the database side.

What I mean is that, we could still have the following:
```javascript
static fields = {
  userName: DataTypes.string(25),
};

// ...

const user = await User.find(1);
console.log(user.userName);
```

But when checking in the database, have a `user_name` field for that.

I will work on this today, I already know how to add this.
Will update this issue asap.

**Comment by eveningkid** [comment]
I just added this behaviour by default. More information in [the PR example](https://github.com/eveningkid/denodb/pull/42#issue-429201844).

I think it's really cleaner this way, very cool! Thanks for the suggestion :)

**Comment by Coteh** [comment]
Damnnn this is so much better! Thank you! 😁

On Jun 6, 2020, at 9:28 AM, Arnaud Dellinger <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #41<https://github.com/eveningkid/denodb/issues/41>.
